laurent22/joplin
# Mobile: Improve dialog styling on large and notched screens

# Summary

This pull request includes a subset of the changes from https://github.com/laurent22/joplin/pull/10465. It:
1. Prevents dialogs from being covered by the device notch when in landscape mode.
2. Vertically centers the plugin panels dialog and plugin info dialogs on large screens.
3. Dims the background of plugin panels and info dialogs.

# Screenshots


## Small screen with notch

![image](https://github.com/laurent22/joplin/assets/46334387/29cbbeae-8d17-4a77-88c8-3d973383b45a)

![image](https://github.com/laurent22/joplin/assets/46334387/3bb21540-0649-4372-9ad5-68507ecb3afc)

<details><summary>Show screenshots from before</summary>

![image](https://github.com/laurent22/joplin/assets/46334387/ba4ee32a-2af0-4e83-b56e-52902d260cb7)


</details>

## Large screen

![image](https://github.com/laurent22/joplin/assets/46334387/cd7e1952-d4c4-4244-bea0-f8b0ce8e5cc5)

<details><summary>Show screenshots from before</summary>

![image](https://github.com/laurent22/joplin/assets/46334387/3359392f-e214-4fbc-8145-2724b70fd498)

</details>

# Testing plan

1. Start with a device in landscape mode.
2. Open the note editor.
3. Add a link using the link dialog.
   - Verify that the link dialog does not overlap the notch (if present).
4. Add a tag using the tag dialog.
   - Verify that the tag dialog does not overlap the notch (if present).
5. Open the plugin panels dialog.
   - Verify that content outside the panels dialog is faded.
   - Verify that the dialog does not overlap the device notch (if present).
6. Close the plugin panels dialog.
7. Repeat steps 2-7 for the same device in portrait mode.

This has been tested on an iOS 17.4 simulator (with a notch) and an Android 14 device (no notch).
